### PR TITLE
How's this?

### PR DESCRIPTION
**Pull request by peterberbec**
So I worked off ozh's random-keyword plugin, but overrode the function, passing the limited charset. Does this work for you? Even after this, I still get:
<code>1lue-	ThinkPad X Series: Our Lightest Thinnest Laptops | Leno[...]</code>

### CODE REVIEW

File: plugin.php [hunk 1]
```diff
@@ -1,17 +1,39 @@
...
 */
 
 
-yourls_add_filter('get_shorturl_charset', 'gfwd_get_shorturl_charset' );
+global $prb_random_keyword;
+/*
+* CONFIG: EDIT THIS
+*/
+/* Length of random keyword */
+$prb_random_keyword['length'] = 5;
```
Comment (cgmckeever) [inline plugin.php]: Does this limit the length of URLs generated?  If I recall, I went down this path, and this sort of sucked, it didnt allow small URLS and then grow on usage, and then, (I cant recall) if it allowed lenght+ sizes ?

Comment (peterberbec) [inline plugin.php]: Major Update! Now I put in an admin page to control the charset, and allows you to specify link length. Neither affect previously made or custom specified links, only new random links.

File: plugin.php [hunk 2]
```diff
@@ -1,17 +1,108 @@
...
+        prb_charset_form ();
+    }
+}
+// Display form to administrate charset list
+function prb_charset_form () {
+    $nonce = yourls_create_nonce( 'form_nonce' ) ;
+    $liste_charset_display = yourls_get_option ( 'charset_liste' );
+    if ( yourls_get_option( 'link_length' ) == false )
+    {	
+		yourls_add_option ( 'link_length', 5 );
```
Comment (peterberbec) [inline plugin.php]: Length defaults to 5 if you don't change it

File: plugin.php [hunk 3]
```diff
@@ -1,17 +1,108 @@
...
+}
+// Display admin page
+function prb_simple_random_charset_do_page () {
+    if( isset( $_POST['charset_form'] ) ) {
+        prb_charset_process ();
+    } else {
+        prb_charset_form ();
+    }
+}
+// Display form to administrate charset list
```
Comment (peterberbec) [inline plugin.php]: Here's the admin page

**Comment by cgmckeever** [comment]
@peterberbec this is great, I feel like this may be best if you release this as a new standalone plugin?  You've put in a lot of work, and its forked significantly enough that you should get full cred?  I know the YOURLs dude will list this on his plug in page?

**Comment by peterberbec** [comment]
ok. I'll let ozh know. I'll leave it forked off your plugin.

**Comment by peterberbec** [comment]
Ok, I've submitted it to ozh and cleaned up the readme and stuff so it's more ready for prime time. Still a fork of your code. It's over at http://berb.ec/y-kcl

Thanks for the tips! It was fun building this. It's my 3rd YOURLS plugin. The first two have less than 20 views. I hope this one gets used, I find it fills an important feature gap.

**Comment by cgmckeever** [comment]
@peterberbec Awesome!  code looks good ... we arent actively using it much now - we were having volume issues and killing the DB when we'd try to make them too quickly :(

**Comment by peterberbec** [comment]
I suggest we move this over here:

https://github.com/peterberbec/yourls-keyword_charset_length/issues/2

**Comment by cgmckeever** [comment]
👍

**Pull request closed without merging**